paimon-common/src/main/java/org/apache/paimon/data/columnar/ColumnarRowIterator.java

```diff
@@ -55,4 +57,13 @@ public InternalRow next() {
             return null;
         }
     }
+
+    public ColumnarRowIterator mapping(
+            @Nullable PartitionInfo partitionInfo, @Nullable int[] indexMapping) {
+        if (!mapped) {
+            rowData.mapping(partitionInfo, indexMapping);
+            mapped = true;
+        }
+        return this;
```

Comment:
Do not return this, return a new `ColumnarRowIterator`.

Reply:
Fixed this.